src/compiler/crystal/macros/methods.cr

```diff
@@ -198,16 +200,18 @@ module Crystal
       macro_raise(node, node.args, self)
     end
 
-    def interpret_read_file(node)
+    def interpret_read_file(node, nilable = false)
       unless node.args.size == 1
-        node.wrong_number_of_arguments "macro call 'read_file'", node.args.size, 1
+        node.wrong_number_of_arguments "macro call '#{node.name}'", node.args.size, 1
       end
 
       node.args[0].accept self
       filename = @last.to_macro_id
-      if File.file?(filename)
+
+      begin
         @last = StringLiteral.new(File.read(filename))
-      else
+      rescue e
```

Comment:
`e` -> `ex` plz

Reply:
Also, one question is if it is a good idea to rescue all possible errors instead of only the specific one that is raised if the file doesn't exist.

Reply:
@yxhuvud I don't see a problem with it since they're re-raised and the only meaningful call (which might raise) is `File.read`.

Reply:
> and the only meaningful call (which might raise) is `File.read`

Yep, this was my reasoning here :slightly_smiling_face: